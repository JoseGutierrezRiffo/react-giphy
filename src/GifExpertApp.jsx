import React, { Fragment, useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";

const GitExpertApp = () => {
  const [categories, setCategories] = useState([]);

  return (
    <Fragment>
      <h2>GitExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </Fragment>
  );
};

export default GitExpertApp;
