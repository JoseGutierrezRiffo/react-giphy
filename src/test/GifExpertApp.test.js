import React from "react";
import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";


describe("prueba con <GifExpertApp />", () => {
  test("debe mostrar el componente", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrar una lista con las categorias", () => {
    const categories = ['Morty', 'Rick', 'Homer']
    const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length)

  });
});
