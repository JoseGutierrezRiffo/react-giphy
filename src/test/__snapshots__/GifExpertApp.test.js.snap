// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba con <GifExpertApp /> debe mostrar el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`prueba con <GifExpertApp /> debe mostrar una lista con las categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Morty"
      key="Morty"
    />
    <GifGrid
      category="Rick"
      key="Rick"
    />
    <GifGrid
      category="Homer"
      key="Homer"
    />
  </ol>
</Fragment>
`;
